lib/core/tools/line_tool/vertex_stack.dart

```diff
@@ -0,0 +1,40 @@
+// Dart imports:
+import 'dart:collection';
+
+// Project imports:
+import 'package:paintroid/core/tools/line_tool/line_tool_vertex.dart';
+
+class VertexStack extends Iterable<Vertex> {
+  final _stack = Queue<Vertex>();
+
+  @override
+  Iterator<Vertex> get iterator => _stack.iterator;
+
+  @override
+  bool get isEmpty => _stack.isEmpty;
+
+  @override
+  Vertex get last => _stack.last;
+
+  void add(Vertex vertex) {
+    _stack.add(vertex);
+  }
+
```

Comment:
Should we also add 
```
void removeLast() {
    _stack.removeLast();
  }
```
Just in case it is needed in the future?

Reply:
This will be needed anyway when implementing 743: undo/redo. I think it makes sense to implement it later in the ticket where it is needed.